completed They were very nice but extremely poor

elka1907 opened the pull request.

elka1907 committed:
completed They were very nice but extremely poor

iBerbasov commented on src/main/java/school/faang/bjs2_89663/Chore.java:
```diff
@@ -0,0 +1,17 @@
+    @Override
+    public void run() {
+        try {
+            System.out.println(Thread.currentThread().getName() + " выполняет задачу: " + chore);
```
Можно начинать пользоваться логгерами, в проекте подключен ломбок - он предоставляет удобный интерфейс @slf4j
В том числе он самостоятельно выводит имя потока, не нужно будет вручную доставать Thread.currentThread().getName()

elka1907 replied:
done

iBerbasov commented on src/main/java/school/faang/bjs2_89663/Chore.java:
```diff
@@ -0,0 +1,17 @@
+            Thread.sleep(1000);
+            System.out.println(chore + " выполнена!");
+        } catch (InterruptedException e) {
+            System.out.println("Задача " + chore + " была прервана.");
```
Тут тоже лучше к логгеру переходить - следует избегать конкатенации строк, чтобы не создавать лишних объектов в памяти
Если не логеры, то лучше форматирование через System.out.printf, String.format или "".formatted()

elka1907 replied:
Done

iBerbasov commented on src/main/java/school/faang/bjs2_89663/WeasleyFamily.java:
```diff
@@ -0,0 +1,28 @@
+        }
+        executor.shutdown();
+        try {
+            if (!executor.awaitTermination(1, TimeUnit.MINUTES)) {
```
Единицу лучше вынести в константы, это антипаттерн магическое число

elka1907 replied:
done

elka1907 committed:
скорректировала:
1. заменила магическое число на константу
2. добавила @Slf4j
3. использовала TimeUnit
